addOns/grpc/src/main/java/org/zaproxy/addon/grpc/internal/VariantGrpc.java

```diff
@@ -0,0 +1,133 @@
+            try {
+                byte[] body = Base64.getDecoder().decode(msg.getRequestBody().getBytes());
+                byte[] payload = DecoderUtils.extractPayload(body);
+                this.protoBufMessageDecoder.decode(payload);
+                this.parseContent(protoBufMessageDecoder.getDecodedToList());
+            } catch (Exception e) {
+                LOGGER.error("Parsing message body failed: {}", e.getMessage());
+            }
+        }
+    }
+
+    public void parseContent(List<String> decodedList) {
+        for (String pair : decodedList) {
+            String[] nameValuePair = pair.split("::", 2);
+            params.add(
+                    new NameValuePair(
+                            NameValuePair.TYPE_JSON,
+                            nameValuePair[0],
+                            nameValuePair[1],
+                            params.size()));
+        }
+    }
+
+    public boolean isValidContentType(String contentType) {
+        return contentType.startsWith("application/grpc-web-text");
+    }
+
+    @Override
+    public List<NameValuePair> getParamList() {
+        return params;
+    }
+
+    @Override
+    public String setParameter(
+            HttpMessage msg, NameValuePair originalPair, String param, String value) {
+        NameValuePair currentPair = params.get(originalPair.getPosition());
+        String reqBody = buildNewBodyContent(currentPair, param, value);
+        try {
+            setEncodedReqBodyMessage(msg, reqBody);
+            return value;
+        } catch (Exception e) {
+            LOGGER.warn("Failed to set parameter in GraphQL message: {}", e.getMessage());
+            return null;
+        }
+    }
+
+    private void setEncodedReqBodyMessage(HttpMessage msg, String newContent) throws Exception {
+        protoBufMessageEncoder.encode(EncoderUtils.parseIntoList(newContent));
+        byte[] encodedMessage = protoBufMessageEncoder.getOutputEncodedMessage();
+        encodedMessage = Base64.getEncoder().encode(encodedMessage);
+        msg.getRequestBody().setBody(encodedMessage);
+    }
+
+    private String buildNewBodyContent(NameValuePair currentPair, String param, String value) {
+        StringBuilder sb = new StringBuilder();
+
+        for (NameValuePair pair : params) {
+            sb.append(pair.getName());
+            sb.append("::");
+            if (pair == currentPair) {
+                String curPairName = currentPair.getName();
+                String[] nameType = curPairName.split(":", 2);
+                if (nameType[1].charAt(0) == '2') sb.append('"').append(value).append('"');
+                else sb.append(value);
+            } else {
+                sb.append(pair.getValue());
+            }
+            sb.append("\n");
+        }
+
+        return sb.toString();
+    }
+
+    @Override
+    public String setEscapedParameter(
+            HttpMessage msg, NameValuePair originalPair, String param, String value) {
+        return "";
```

Comment:
Should behave the same as `setParameter` if not able to do any "escaping", although in this context it would set the value directly without encoding the field (tricky?).

Reply:
set the value without encoding but why ? i thought i should call setParameter(). and can you please tell me use of this setEscapedParameter and where we should call it

Reply:
This method allows the scan rules to set their own "raw" values, this is usually done for rules that test for specific behaviours (e.g. handling of malformed payloads). We don't call it here, it's the scan rules that would call it.

Reply:
but even the setParameter is call by scan rules, and i search setEscapedParameter in ascanrules, but no scan rule is calling it

Reply:
Yes, no rules in this codebase is calling it, that our codebase does not do something does not mean there's no use case or no one is calling it (there are, even last week(?) a user needed this method for a custom scan rule).

Reply:
okay, this method should still call setParameter method ?

Reply:
If we are not able to inject the value without escaping, yes, it should just call the other method.

Reply:
Actually sir, i didn't understand the use of this method fully, can you please give me some examples and what escaping means here ?

Reply:
"Escaping" in this context means adjusting the value being set to conform to the format of the `Variant` (e.g. for URL means URL encoding; for JSON, escaping as string), which the `setParameter` does, it will set a value that still needs to be "escaped". The `setEscapedParameter` allows to set an already "escaped" value, so the variant should just insert it in the whole payload without "escaping" it. Is that clearer?

Reply:
Thank you for such a good explanation, I get it, setEscapedParameter just put the value in response body without any checking or conversion according to variant, but in gRPC we need to encode the value, we can't send it unencoded, so we can just call setParameter method from here

Reply:
In theory we should be able to at least insert the payload in the expected location, but for now it can just call the `setParameter`as is and add JavaDoc mentioning that it just calls the other method.

Reply:
I see in one of the pushes it's already calling the method so just needs the JavaDoc.

Reply:
is this good ?
```
/**
 * Sets an escaped parameter in the HTTP message.
 *
 * @param msg The HTTP message to modify.
 * @param originalPair The original name-value pair.
 * @param param The parameter name.
 * @param value The parameter value.
 * @return The modified parameter.
 */

  @Override
    public String setEscapedParameter(
            HttpMessage msg, NameValuePair originalPair, String param, String value) {
        return setParameter(msg, originalPair, param, value);
    }
```

Reply:
Just needs to say that calls the other method, e.g.:
```java
/**
 * Calls {@link setEscapedParameter}.
 */
```
(adjust as need to be valid link.)